src/rust/ide/src/view/node_editor.rs

```diff
@@ -167,6 +173,21 @@ impl GraphEditorIntegratedWithController {
                 }
             }));
         }
+
+
+        // === Project Renaming ===
+
+        let project_name = &model.editor.project_name;
+        frp::extend! {network
+            eval project_name.frp.outputs.name((name) {
+                model.rename_project(name.to_string());
```

Comment:
why this api requires allocation?

Reply:
@wdanilo because `name` is moved to an async closure inside of `rename_project`.

Reply:
Hmm, I still do not understand it. No matter what happens inside `rename_project`, why its signature is designed to force passing the ownership ?